packages/crypto/src/turnkey.ts

```diff
@@ -403,3 +405,62 @@ export const encryptWalletToBundle = async ({
   const privateKeyBundle = hpkeEncrypt({ plainTextBuf, targetKeyBuf });
   return formatHpkeBuf(privateKeyBundle);
 };
+
+/**
+ * Verifies that a **session JWT** was signed by Turnkey’s
+ * notarizer key (P-256 / ES256, compact 64-byte r‖s signature).
+ *
+ * How it works
+ * ------------
+ * 1.  Split the JWT into `header.payload.signature`.
+ * 2.  **Double-hash** the string `"header.payload"`:
+ *        `h1 = sha256(header.payload)`
+ *        `msg = sha256(h1)`
+ *     (The Rust signer feeds `h1` into `SigningKey::sign`, which hashes once
+ *     more internally, yielding `msg`.)
+ * 3.  Base64-URL-decode the signature (`r||s`, 64 bytes).
+ * 4.  Import the notarizer public key (hex `04‖X‖Y` → `Uint8Array`).
+ * 5.  Call `p256.verify(signature, msg, publicKey)`; noble treats the 32-byte
+ *     `msg` as a pre-hashed digest and performs ECDSA verification.
+ *
+ * @param jwt   The session JWT to validate.
+ * @param dangerouslyOverrideNotarizerPublicKey *(optional)* Hex-encoded
+ *              uncompressed P-256 public key to verify against (use only in
+ *              tests).  Defaults to the production notarizer key.
+ * @returns `true` if the signature is valid for the given key, else `false`.
+ * @throws  If the JWT is malformed.
+ */
+export const verifySessionJwtSignature = async (
+  jwt: string,
+  dangerouslyOverrideNotarizerPublicKey?: string,
+): Promise<boolean> => {
+  const notarizerKeyHex =
+    dangerouslyOverrideNotarizerPublicKey ?? PRODUCTION_NOTARIZER_PUBLIC_KEY;
+
+  /* 1. split JWT -------------------------------------------------------- */
+  const [headerB64, payloadB64, signatureB64] = jwt.split(".");
+  if (!signatureB64) throw new Error("invalid JWT: need 3 parts");
```

Comment:
should we also error if `payloadB64` is somehow unpopulated as well?

Reply:
nvm, if `payloadB64` is unpopulated, then `signatureB64` will def be unpopulated as well. so this single check should be sufficient